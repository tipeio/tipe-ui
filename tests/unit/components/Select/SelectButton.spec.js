import { shallowMount } from '@vue/test-utils'
import { createRenderer } from 'vue-server-renderer'
import selectTests from './selectTests'

import TipeSelectButton from '@/components/Select/SelectButton'

const options = [
  { label: 'Owner', value: 'owner' },
  { label: 'Admin', value: 'admin' }
]

const props = {
  options
}
describe('TipeSelectButton.vue', () => {
  it('renders', () => {
    const renderer = createRenderer()
    const wrapper = shallowMount(TipeSelectButton, {
      propsData: {
        ...props
      }
    })
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
  it('has correct data-tipe-ui attibute', () => {
    const wrapper = shallowMount(TipeSelectButton, {
      propsData: {
        ...props
      }
    })
    expect(wrapper.attributes()['data-tipe-ui']).toBe('TipeSelectButton')
  })
  selectTests(TipeSelectButton)
})
