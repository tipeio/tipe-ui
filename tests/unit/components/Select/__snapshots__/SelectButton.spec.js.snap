// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TipeSelectButton.vue renders 1`] = `
<div data-tipe-ui="TipeSelectButton" class="select-flex">
  <div class="select">
    <TipeNativeSelect-stub options="[object Object],[object Object]" placeholder="" selectedValue="[object Object]"></TipeNativeSelect-stub>
    <TipeButton-stub color="primary" iconBefore="" iconAfter="arrow-down" name="" size="medium" tabindex="0">Owner</TipeButton-stub>
    <!---->
  </div>
</div>
`;

exports[`TipeSelectButton.vue renders without options 1`] = `
<div data-tipe-ui="TipeSelectButton" class="select-flex">
  <div class="select">
    <TipeNativeSelect-stub placeholder="" selectedValue="[object Object]"></TipeNativeSelect-stub>
    <TipeButton-stub color="primary" iconBefore="" iconAfter="arrow-down" name="" size="medium" tabindex="0">Please provide select options</TipeButton-stub>
    <!---->
  </div>
</div>
`;
